modules: memfault-firmware-sdk: migrate coredump from PM to devicetree (#28820)
nrfconnect/sdk-nrf: doc/nrf/libraries/debug/memfault_ncs.rst

* :kconfig:option:`CONFIG_MEMFAULT_NCS_STACK_METRICS`
* :kconfig:option:`CONFIG_MEMFAULT_NCS_BT_METRICS`

.. _snippet-memfault-coredump:

Flash-backed coredump storage
-----------------------------

When :kconfig:option:`CONFIG_MEMFAULT_NCS_INTERNAL_FLASH_BACKED_COREDUMP` is enabled, the storage region is resolved at build time using one of the following sources:

* **Partition Manager** (legacy): When :kconfig:option:`CONFIG_PARTITION_MANAGER_ENABLED` is ``y``, a ``MEMFAULT_STORAGE`` partition is generated automatically. The size defaults to 64 kB and can be overridden with :kconfig:option:`CONFIG_PM_PARTITION_SIZE_MEMFAULT_STORAGE`.

> * **Devicetree** (NCS 3.4+ recommended): When Partition Manager is disabled, the backend locates a fixed partition labelled ``memfault_coredump_partition`` under the internal flash controller.

Reviewer: ```suggestion
* **Partition Manager** (legacy): When :kconfig:option:`CONFIG_PARTITION_MANAGER_ENABLED` is ``y``, a ``MEMFAULT_STORAGE`` partition is generated automatically. 
  The size defaults to 64 kB, and you can override it with :kconfig:option:`CONFIG_PM_PARTITION_SIZE_MEMFAULT_STORAGE`.
* **Devicetree** (|NCS| v3.4.0 or later recommended): When Partition Manager is disabled, the backend locates a fixed partition labeled ``memfault_coredump_partition`` under the internal flash controller.
```
Please change labelled -> labeled, wherever applicable.

* :kconfig:option:`CONFIG_MEMFAULT_NCS_STACK_METRICS`
* :kconfig:option:`CONFIG_MEMFAULT_NCS_BT_METRICS`

.. _snippet-memfault-coredump:

Flash-backed coredump storage
-----------------------------

> When :kconfig:option:`CONFIG_MEMFAULT_NCS_INTERNAL_FLASH_BACKED_COREDUMP` is enabled, the storage region is resolved at build time using one of the following sources:

Reviewer 2: ```suggestion
When the :kconfig:option:`CONFIG_MEMFAULT_NCS_INTERNAL_FLASH_BACKED_COREDUMP` Kconfig option is enabled, the storage region is resolved at build time using one of the following sources:
```

* **Partition Manager** (legacy): When :kconfig:option:`CONFIG_PARTITION_MANAGER_ENABLED` is ``y``, a ``MEMFAULT_STORAGE`` partition is generated automatically.
  The size defaults to 64 kB, and you can override it with :kconfig:option:`CONFIG_PM_PARTITION_SIZE_MEMFAULT_STORAGE`.
* **Devicetree** (|NCS| v3.4.0 or later recommended): When Partition Manager is disabled, the backend locates a fixed partition labeled ``memfault_coredump_partition`` under the internal flash controller.

> The easiest way to enable the devicetree path is the :ref:`snippet-memfault-coredump` snippet, which supplies a per-board overlay defining the partition:

Reviewer 2: ```suggestion
The easiest way to enable the devicetree path is the :ref:`snippet-memfault-coredump` snippet that supplies a board-specific overlay defining the partition:
```

.. code-block:: console

   west build -b <board> -S memfault-coredump <application>

For boards not covered by the snippet, add a node similar to the following to your application overlay:

.. code-block:: dts

   &flash0 {
       partitions {
           memfault_coredump_partition: partition@fc000 {

>                label = "memfault_coredump_partition";

Reviewer 3: missing compatible

Author: fixed!
